World.py

```diff
@@ -187,6 +191,158 @@ def __init__(self, id, settings):
+                        b.goals.append(Goal(self, { 'replace': 'Light Medallion' }, 'Light Blue', items=[{'name': 'Light Medallion','quantity': 1,'minimum': 1,'hintable': True}]))
+                        b.minimum_goals = self.settings.bridge_medallions if self.settings.bridge == 'medallions' else self.settings.bridge_rewards
+                    if self.settings.bridge == 'vanilla':
+                        b.goals.append(Goal(self, { 'replace': 'Shadow Medallion' }, 'Pink', items=[{'name': 'Shadow Medallion','quantity': 1,'minimum': 1,'hintable': True}]))
+                        b.goals.append(Goal(self, { 'replace': 'Spirit Medallion' }, 'Yellow', items=[{'name': 'Spirit Medallion','quantity': 1,'minimum': 1,'hintable': True}]))
+                        min_goals = 2
+                        if not 'Light Arrows' in self.item_added_hint_types['always']:
+                            if self.settings.item_pool_value == 'plentiful':
+                                arrows = 2
+                            else:
+                                arrows = 1
+                            b.goals.append(Goal(self, 'Evil\'s Bane', 'Light Blue', items=[{'name': 'Light Arrows','quantity': arrows,'minimum': 1,'hintable': True}]))
+                            min_goals += 1
+                        b.minimum_goals = min_goals
+                    b.goal_count = len(b.goals)
+                    if (self.settings.bridge_tokens > 0 and self.settings.bridge == 'tokens' and (self.settings.shuffle_ganon_bosskey != 'on_lacs' or self.settings.lacs_condition != 'tokens' or self.settings.bridge_tokens >= self.settings.lacs_tokens)):
+                        b.goals.append(Goal(self, 'Skulls', 'Light Blue', items=[{'name': 'Gold Skulltula Token','quantity': 100,'minimum': self.settings.bridge_tokens,'hintable': False}]))
+                        b.goal_count = round(self.settings.bridge_tokens / 10)
+                        b.minimum_goals = 1
+                    self.goal_categories[b.name] = b
+
+                if self.settings.shuffle_ganon_bosskey == 'on_lacs':
+                    if ((self.settings.lacs_stones > 0 and self.settings.lacs_condition == 'stones' and (self.settings.lacs_stones > self.settings.bridge_stones or self.settings.bridge != 'stones')) or
+                    (self.settings.lacs_rewards > 0 and self.settings.lacs_condition == 'dungeons' and
+                    ((self.settings.lacs_rewards > self.settings.bridge_medallions and self.settings.bridge == 'medallions') or
+                    (self.settings.lacs_rewards > self.settings.bridge_stones and self.settings.bridge == 'stones') or
+                    (self.settings.lacs_rewards > self.settings.bridge_rewards and self.settings.bridge == 'dungeons') or
+                    not (self.settings.lacs_rewards <= 2 and self.settings.bridge == 'vanilla')))):
```

Comment:
```suggestion
                    if ((self.settings.lacs_stones > 0
                         and self.settings.lacs_condition == 'stones'
                         and (self.settings.lacs_stones > self.settings.bridge_stones or self.settings.bridge != 'stones'))
                        or (self.settings.lacs_rewards > 0
                            and self.settings.lacs_condition == 'dungeons'
                            and ((self.settings.lacs_rewards > self.settings.bridge_medallions and self.settings.bridge == 'medallions')
                                 or (self.settings.lacs_rewards > self.settings.bridge_stones and self.settings.bridge == 'stones')
                                 or (self.settings.lacs_rewards > self.settings.bridge_rewards and self.settings.bridge == 'dungeons')
                                 or self.settings.lacs_rewards > 2
                                 or self.settings.bridge != 'vanilla'))):
```

Reply:
Hm? I thought my suggestion just simplified the expression `not (a and b)` into `not a or not b` while I was indenting everything. Was your original code wrong then? (Is this resolved?)